--- a/homeassistant/homeassistant/components/kodi/media_player.py	2019-08-03 13:26:04.012756909 -0700
+++ b/homeassistant/homeassistant/components/kodi/media_player.py	2019-08-03 13:31:03.877703759 -0700
@@ -590,9 +590,16 @@
         return self._name
 
     @property
+    def is_connected(self):
+        """Return True if the entity requires manual updating."""
+        if not self._enable_websocket:
+            return False
+        return self._ws_server.connected
+
+    @property
     def should_poll(self):
         """Return True if entity has to be polled for state."""
-        return not (self._enable_websocket and self._ws_server.connected)
+        return True
 
     @property
     def volume_level(self):

--- a/homeassistant/homeassistant/components/kodi/__init__.py   2019-08-03 13:26:04.012756909 -0700
+++ b/homeassistant/homeassistant/components/kodi/__init__.py   2019-08-03 13:31:03.877703759 -0700
@@ -77,7 +77,7 @@
             await getattr(player, method["method"])(**params)
 
         for player in target_players:
-            if player.should_poll:
+            if not player.is_connected:
                 update_coro = player.async_update_ha_state(True)
                 update_tasks.append(update_coro)
 
